assets/blocks/lesson-actions/complete-lesson-block/index.js

```diff
@@ -25,6 +25,9 @@ export default createButtonBlockType( {
 			text: {
 				default: __( 'Complete Lesson', 'sensei-lms' ),
 			},
+			buttonClassName: {
+				default: [ 'sensei-stop-double-submission' ],
```

Comment:
> This should not be added based on an editable block attribute, since it's a technical class that always needs to be present. The rendering logic in PHP would be a better place for it.

I would like to avoid regexp replacements if possible. Ideally this should be placed in the save function but the complexity is that the button is shared and we only want to add it in cases where the button is part of form.

Reply:
We could think about moving the buttons to be fully PHP-rendered based on the attributes, instead of the current regexp approach ([class-sensei-block-take-course.php#L93](https://github.com/Automattic/sensei/blob/e65e0d539203a6c833bcc9737f6d2e596d205589/includes/blocks/class-sensei-block-take-course.php#L93)) for changing the HTML generated by the save function.

Reply:
> We could think about moving the buttons to be fully PHP-rendered based on the attributes

We could but won't this be duplicated work? We used a regexp previously as the block was already released and we would need versioning. I also think that being able to add a class to the button is a useful attribute and it could be also used in the future.